bitsandbytes/cuda_specs.py

```diff
@@ -0,0 +1,41 @@
+import dataclasses
+from typing import List, Optional, Tuple
+
+import torch
+
+
+@dataclasses.dataclass(frozen=True)
```

Comment:
also, super nice and clean, the whole module! Really like how you factor out everything, much easier to follow and maintain now